buildchain/buildchain/salt_tree.py

```diff
@@ -216,6 +216,10 @@ def _get_parts(self) -> Iterator[str]:
         renderer=targets.Renderer.JSON,
     ),
 
+    Path('salt/metalk8s/addons/dex/deployed/init.sls'),
+    Path('salt/metalk8s/addons/dex/deployed/chart.sls'),
+    Path('salt/metalk8s/addons/dex/deployed/namespace.sls'),
+
```

Comment:
This is not related to the commit message, which states two things:

- "salt: deploy dex" - I would expect this commit to actually **deploy** Dex using Salt, not just prepare the buildchain
- "This commit adds the dex image to the buildchain which is required for dex deployment" - Then I don't expect anything related to Salt states in there

I think we could settle for a single commit, that handles:

- adding required images in the buildchain
- rendering `chart.sls`
- adding the few dependencies (like `tls-secret.sls`)
- adding those files in the buildchain `salt_tree.py`

The whole "Deploy Dex" feature is thus self-contained in this commit. Fixing the render script or other unrelated stuff should go in other commits (if required by this one, then go before it, if not, after).